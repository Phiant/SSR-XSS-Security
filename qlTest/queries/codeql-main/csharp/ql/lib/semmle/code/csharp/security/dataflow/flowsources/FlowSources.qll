/** Provides classes representing various flow sources for taint tracking. */

private import semmle.code.csharp.dataflow.internal.ExternalFlow
private import codeql.threatmodels.ThreatModels
import semmle.code.csharp.security.dataflow.flowsources.Remote
import semmle.code.csharp.security.dataflow.flowsources.Local
import semmle.code.csharp.security.dataflow.flowsources.Stored

/**
 * A data flow source.
 */
abstract class SourceNode extends DataFlow::Node {
  /**
   * Gets a string that represents the source kind with respect to threat modeling.
   */
  abstract string getThreatModel();

  /** Gets a string that describes the type of this flow source. */
  abstract string getSourceType();
}

/**
 * A class of data flow sources that respects the
 * current threat model configuration.
 */
class ThreatModelFlowSource extends DataFlow::Node {
  ThreatModelFlowSource() {
    exists(string kind |
      // Specific threat model.
      currentThreatModel(kind) and
      (this.(SourceNode).getThreatModel() = kind or sourceNode(this, kind))
    )
  }
}
