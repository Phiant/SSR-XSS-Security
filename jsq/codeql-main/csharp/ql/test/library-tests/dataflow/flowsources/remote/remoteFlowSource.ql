import semmle.code.csharp.security.dataflow.flowsources.Remote

from RemoteFlowSource source
where source.getLocation().getFile().fromSource()
select source, source.getSourceType()
