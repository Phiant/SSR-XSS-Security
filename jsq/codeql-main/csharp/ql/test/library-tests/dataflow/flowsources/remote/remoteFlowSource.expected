| Controller.cs:11:43:11:52 | sampleData | ASP.NET MVC action method parameter |
| Controller.cs:11:62:11:66 | taint | ASP.NET MVC action method parameter |
| Controller.cs:16:43:16:52 | sampleData | ASP.NET MVC action method parameter |
| Controller.cs:16:62:16:66 | taint | ASP.NET MVC action method parameter |
| RemoteFlowSource.cs:30:17:30:23 | access to parameter request | ASP.NET query string |
| RemoteFlowSource.cs:31:17:31:23 | access to parameter textBox | ASP.NET user input |
| RemoteFlowSource.cs:34:44:34:53 | sampleData | web service input |
| RemoteFlowSource.cs:34:63:34:67 | taint | web service input |
| RemoteFlowSource.cs:40:17:40:23 | access to parameter request | ASP.NET query string |
| RemoteFlowSource.cs:45:17:45:23 | access to parameter request | ASP.NET query string |
| RemoteFlowSource.cs:45:17:45:42 | access to property RawUrl | ASP.NET unvalidated request data |
